File: android/app/src/main/java/jagex3/graphics2/hw/NativeInterface.java
package jagex3.graphics2.hw;


public final class NativeInterface {

    public long peer;

    public NativeInterface(int arg0, int arg1) {
        this.init(arg0, arg1);
    }

    public void copyNormals(short[] arg0, short[] arg1, short[] arg2, byte[] arg3, float arg4, float arg5, int arg6, int arg7, int arg8, long arg9) {}

    public void release() {}

    public void copyLighting(short[] arg0, byte[] arg1, short[] arg2, short[] arg3, short[] arg4, short[] arg5, byte[] arg6, int arg7, int arg8, short[] arg9, int arg10, int arg11, int arg12, long arg13) {}

    public void init(int arg0, int arg1) {}

    public void initTextureMetrics(int arg0, byte arg1, byte arg2) {}

    public void setSunColour(float arg0, float arg1, float arg2, float arg3, float arg4) {}

    public void setSunDirection(float arg0, float arg1, float arg2) {}

    public void copyPositions(int[] arg0, int[] arg1, int[] arg2, short[] arg3, int arg4, int arg5, int arg6, long arg7) {}

    public void copyTexCoords(float[] arg0, float[] arg1, int arg2, int arg3, int arg4, long arg5) {}

    public void setAmbient(float arg0) {}

    public void copyColours(short[] arg0, byte[] arg1, short[] arg2, int arg3, short[] arg4, int arg5, int arg6, int arg7, long arg8) {}
}

File: src/jvmMain/kotlin/jagex3/graphics2/hw/NativeInterface.kt
package jagex3.graphics2.hw

actual class NativeInterface actual constructor(arg0: Int, arg1: Int) {
    /*private*/ actual val peer: Long = 0

    init {
        this.init(arg0, arg1)
    }

    actual external fun copyLighting(arg0: ShortArray?, arg1: ByteArray?, arg2: ShortArray?, arg3: ShortArray?, arg4: ShortArray?, arg5: ShortArray?, arg6: ByteArray?, arg7: Int, arg8: Int, arg9: ShortArray?, arg10: Int, arg11: Int, arg12: Int, arg13: Long)

    /*private*/ actual external fun init(arg0: Int, arg1: Int)

    actual external fun copyNormals(arg0: ShortArray?, arg1: ShortArray?, arg2: ShortArray?, arg3: ByteArray?, arg4: Float, arg5: Float, arg6: Int, arg7: Int, arg8: Int, arg9: Long)

    actual external fun copyPositions(arg0: IntArray?, arg1: IntArray?, arg2: IntArray?, arg3: ShortArray?, arg4: Int, arg5: Int, arg6: Int, arg7: Long)

    actual external fun setSunColour(arg0: Float, arg1: Float, arg2: Float, arg3: Float, arg4: Float)

    actual external fun initTextureMetrics(arg0: Int, arg1: Byte, arg2: Byte)

    actual external fun release()

    actual external fun setAmbient(arg0: Float)

    actual external fun copyTexCoords(arg0: FloatArray?, arg1: FloatArray?, arg2: Int, arg3: Int, arg4: Int, arg5: Long)

    actual external fun copyColours(arg0: ShortArray?, arg1: ByteArray?, arg2: ShortArray?, arg3: Int, arg4: ShortArray?, arg5: Int, arg6: Int, arg7: Int, arg8: Long)

    actual external fun setSunDirection(arg0: Float, arg1: Float, arg2: Float)
}
